lib/rgeo/geos/ffi_feature_methods.rb

```
@@ -97,11 +97,9 @@ def envelope
       end
 
       def boundary
-        if self.class == FFIGeometryCollectionImpl
-          nil
-        else
-          @factory.wrap_fg_geom(@fg_geom.boundary, nil)
-        end
+        @factory.wrap_fg_geom(@fg_geom.boundary, nil)
+      rescue ::Geos::GEOSException
+        raise Error::InvalidGeometry, "Operation not supported by GeometryCollection"
```

We could raise `UnsupportedOperation` however this would make it inconsistant with the CAPI. And the CAPI workflow is not able to guess we're in that specific context without asserting even more from error strings.

---

I think making it consistent with CAPI is better in this case.